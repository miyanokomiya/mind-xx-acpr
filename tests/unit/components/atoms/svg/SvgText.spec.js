import { mount } from '@vue/test-utils'
import Target from '@/components/atoms/svg/SvgText.vue'

describe('components/atoms/svg/SvgText.vue', () => {
  describe('snapshot', () => {
    it('does not have link', () => {
      const wrapper = mount(Target, {
        propsData: {
          text: 'prop-text',
          x: 1,
          y: 2,
          fontSize: 10,
          fill: 'red',
          fontWeight: 20,
          textDecoration: 'prop-textDecoration',
        },
      })
      expect(wrapper.element).toMatchSnapshot()
    })
    it('has link', () => {
      const wrapper = mount(Target, {
        propsData: {
          text: 'http://example.com',
        },
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('methods', () => {
    describe('getBBox', () => {
      it('returns bbox of svg node', () => {
        const wrapper = mount(Target, {
          propsData: {
            text: 'http://example.com',
          },
        })
        const bbox = wrapper.vm.getBBox()
        expect(bbox.width).toBeGreaterThan(0)
        expect(bbox.height).toBeGreaterThan(0)
      })
    })

    describe('moveLink', () => {
      const orgOpen = window.open
      beforeEach(() => (window.open = jest.fn()))
      afterEach(() => (window.open = orgOpen))

      it('exec window.open if text is link', () => {
        const wrapper = mount(Target, {
          propsData: {
            text: 'http://example.com',
          },
        })
        wrapper.vm.moveLink()
        expect(window.open.mock.calls).toEqual([['http://example.com']])
      })
      it('not exec window.open if text is not link', () => {
        const wrapper = mount(Target, {
          propsData: {
            text: 'text text',
          },
        })
        wrapper.vm.moveLink()
        expect(window.open).toHaveBeenCalledTimes(0)
      })
    })
  })
})
