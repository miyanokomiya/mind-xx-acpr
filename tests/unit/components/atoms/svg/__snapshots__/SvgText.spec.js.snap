// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/atoms/svg/SvgText.vue snapshot does not have link 1`] = `
<text
  fill="red"
  font-size="10"
  font-weight="20"
  text-decoration="prop-textDecoration"
  x="1"
  y="2"
>
  
    prop-text
  
</text>
`;

exports[`components/atoms/svg/SvgText.vue snapshot has link 1`] = `
<text
  fill="black"
  font-size="10"
  font-weight="500"
  text-decoration=""
  x="0"
  y="0"
>
  <a
    style="cursor: pointer; text-shadow: 1px 1px 0 blue;"
    target="_blank"
    xlink:href="http://example.com"
  >
    
    http://example.com
  
  </a>
</text>
`;
